lstchain/image/muon/muon_analysis.py

```diff
@@ -87,19 +91,17 @@ def fit_muon(x, y, image, geom, tailcuts):
     image_clean = image * clean_mask
     muonringparam = muonring.fit(x, y, image_clean)
 
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2)
-                       + np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
-        )
-
-    dist = np.sqrt(np.power(x - muonringparam.ring_center_x, 2) +
-                       np.power(y - muonringparam.ring_center_y, 2))
-    ring_dist = np.abs(dist - muonringparam.ring_radius)
-
-    muonringparam = muonring.fit(
-            x, y, image_clean * (ring_dist < muonringparam.ring_radius * 0.4)
+    max_allowed_outliers_distance = 0.4
+    # Do an iterative fit removing pixels which are beyond max_allowed_outliers_distance*ring_radius
+    # of the ring (along the radial direction):
+    # The goal is to improve fit for good rings with very few additional non-ring bright pixels.
+
+    for _ in [0]*2:  # just to iterate the fit twice more
```

Comment:
```

In [1]: %%timeit 
   ...: for i in range(2): 
   ...:     pass 
   ...:                                                                                                                                       
153 ns ± 1.83 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [2]: %%timeit 
   ...: for i in [0] * 2: 
   ...:     pass 
   ...:                                                                                                                                       
91 ns ± 1.21 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %%timeit 
   ...: for i in (0, 0): 
   ...:     pass 
   ...:                                                                                                                                       
38 ns ± 0.59 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

However, this will not play any role in the overall runtime of this analysis chain.